src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -55,7 +55,8 @@ public static class DependencyGraphRestoreUtility
                         sources,
                         dgSpec,
                         parentId,
-                        forceRestore);
+                        forceRestore,
+                        isRestore: true);
```

Comment:
This is very confusing and very error prone. 

It's super easy to end up using the wrong API here as they are simply different overloads of the same method.

Reply:
hmm agree it might be confusing for lot of people. Let me see if I can simplify it.